src/botPage/view/blockly/blocks/shared.js

```diff
@@ -20,7 +20,9 @@ export const updatePurchaseChoices = (contractType, oppositesName) => {
     purchases.forEach(purchase => {
         const value = purchase.getField('PURCHASE_LIST').getValue();
         Blockly.WidgetDiv.hideIfOwner(purchase.getField('PURCHASE_LIST'));
-        if (value === purchaseChoices[0][1]) {
+        if (!purchaseChoices.length) {
```

Comment:
The purchase choices should never be empty, their default value is `[[translate('Click to select'), '']]`